### PR TITLE
[feat] adapt to use cred-based authentication for smtp

### PR DESCRIPTION
**Pull request by kalikaneko**

### CODE REVIEW

**Comment by kalikaneko** [comment]
this needs https://github.com/leapcode/leap_mail/pull/217

File: src/leap/bitmask/services/mail/smtpbootstrapper.py [hunk 1]
```diff
@@ -138,19 +136,25 @@ def _start_smtp_service(self):
         client_cert_path = self._smtp_config.get_client_cert_path(
             self._userid, self._provider_config, about_to_download=True)
 
-        from leap.mail.smtp import setup_smtp_gateway
+        # XXX this should be defined in leap.mail.smtp, it's in bitmask.core
+        # right now.
+        SendmailOpts = namedtuple(
+            'SendmailOpts', ['cert', 'key', 'hostname', 'port'])
```
Comment (meskio) [inline src/leap/bitmask/services/mail/smtpbootstrapper.py]: Is there any reason to don't define `SendmailOpts` in leap.mail now? Looks weird that leap.mail uses it, but doesn't define it anywhere.

File: src/leap/bitmask/backend/components.py [hunk 2]
```diff
@@ -804,9 +804,19 @@ def register_imap_token(imap_token):
                     self._signaler.soledad_got_service_token,
                     ('imap', imap_token))
 
+        def register_smtp_token(smtp_token):
+            self._service_tokens['smtp'] = smtp_token
+            if self._signaler is not None:
+                self._signaler.signal(
+                    self._signaler.soledad_got_service_token,
+                    ('smtp', smtp_token))
```
Comment (meskio) [inline src/leap/bitmask/backend/components.py]: Maybe we can converge `register_imap_token` and `register_smtp_token` into one function adding a parameter `service` or something to it.

File: src/leap/bitmask/services/mail/smtpbootstrapper.py (hunk 1 above)
Comment (kalikaneko) [inline src/leap/bitmask/services/mail/smtpbootstrapper.py]: yes. I want to move in block many of the things that are defined in bitmask.core.mail_services right now.

and all this bootstrapping is going to be deprecated soon all the same, being moved to bonafide.

File: src/leap/bitmask/backend/components.py (hunk 2 above)
Comment (kalikaneko) [inline src/leap/bitmask/backend/components.py]: done

**Pull request merged**